arangod/RocksDBEngine/RocksDBMetadata.cpp

```diff
@@ -416,11 +416,13 @@ Result RocksDBMetadata::serializeMeta(rocksdb::WriteBatch& batch,
 
   // Step 4. store the revision tree
   if (rcoll->needToPersistRevisionTree(maxCommitSeq)) {
+    output.clear();
+      
+    rocksdb::SequenceNumber seq =
+        rcoll->serializeRevisionTree(output, maxCommitSeq, force);
+    appliedSeq = std::min(appliedSeq, seq);
+
     if (coll.useSyncByRevision()) {
```

Comment:
If `needToPersistRevisionTree` returns true, then `useSyncByRevision` will always return `true`. So we can leave out the check here and just assert it.